Allow GOAWAY on request streams to migrate individual requests (#1617)
moq-wg/moq-transport: draft-ietf-moq-transport.md

  New Session URI Length (vi64),
  New Session URI (..),
  Timeout (vi64),

>   [Request ID (vi64)],

Reviewer: I think we should just remove Request ID.  I really don't think it's useful.

Reviewer: I wrote #1623

Author: This is in a separate PR and whoever loses the merge race can address

}
~~~
{: #moq-transport-goaway-format title="MOQT GOAWAY Message"}

* New Session URI: When received by a client, indicates where the client can
  connect to continue this session or re-issue this request.  The client MUST
  use this URI for the new session if provided. If the URI is zero bytes long,
  the current URI is reused instead. The new session URI SHOULD use the same scheme
  as the current URI to ensure compatibility.  The maximum length of the New
  Session URI is 8,192 bytes.  If an endpoint receives a length exceeding the
  maximum, it MUST close the session with a `PROTOCOL_VIOLATION`.

  If a server receives a GOAWAY with a non-zero New Session URI Length it MUST
  close the session with a `PROTOCOL_VIOLATION`.

* Timeout: The time in milliseconds the sender will wait for graceful closure.
  When sent on the control stream, the sender closes the session with
  `GOAWAY_TIMEOUT` after the indicated timeout if there are still open requests.
  When sent on a request stream, the sender SHOULD reset the stream with
  `GOING_AWAY` after the indicated timeout.  A value of 0 indicates the sender has no

>   specific timeout, and the recipient SHOULD still migrate as quickly as

Reviewer: ```suggestion
  specific timeout, but the recipient SHOULD migrate as quickly as
```

  possible. This is a hint; the sender of the GOAWAY MAY close the session or

>   reset the stream before the indicated timeout has elapsed.

Reviewer: ```suggestion
  reset the request stream before the indicated timeout has elapsed.
```
